corehq/apps/userreports/dbaccessors.py

```diff
@@ -13,25 +15,17 @@ def get_number_of_report_configs_by_data_source(domain, data_source_id):
     ).one()['value']
 
 
+@unit_testing_only
 def get_all_report_configs():
-    from corehq.apps.userreports.models import ReportConfiguration
-    ids = [res['id'] for res in ReportConfiguration.view(
-        'userreports/report_configs_by_domain',
-        reduce=False,
-        include_docs=False,
-    )]
-    for result in iter_docs(ReportConfiguration.get_db(), ids):
-        yield ReportConfiguration.wrap(result)
+    all_domains = Domain.get_all()
+    for domain in all_domains:
+        for report_config in get_report_configs_for_domain(domain.name):
+            yield report_config
```

Comment:
I think this is the same anti-pattern we talked about earlier; I think you can fix like you did in https://github.com/dimagi/commcare-hq/pull/9244